plugin-server/src/cdp/consumers/cdp-cyclotron-plugins-worker.consumer.ts

```diff
@@ -0,0 +1,193 @@
+import { ProcessedPluginEvent, RetryError } from '@posthog/plugin-scaffold'
+import { DateTime } from 'luxon'
+
+import { Response, trackedFetch } from '~/src/utils/fetch'
+import { status } from '~/src/utils/status'
+
+import { PLUGINS_BY_ID } from '../legacy-plugins'
+import { LegacyPluginLogger, LegacyPluginMeta } from '../legacy-plugins/types'
+import { sanitizeLogMessage } from '../services/hog-executor.service'
+import { HogFunctionInvocation, HogFunctionInvocationResult, HogFunctionTypeType } from '../types'
+import { CdpCyclotronWorker } from './cdp-cyclotron-worker.consumer'
+
+type PluginState = {
+    setupPromise: Promise<any>
+    errored: boolean
+    meta: LegacyPluginMeta
+}
+
+/**
+ * NOTE: This is a consumer to take care of legacy plugins.
+ */
+export class CdpCyclotronWorkerPlugins extends CdpCyclotronWorker {
+    protected name = 'CdpCyclotronWorkerPlugins'
+    protected queue = 'plugin' as const
+    protected hogTypes: HogFunctionTypeType[] = ['destination']
+
+    private pluginState: Record<string, PluginState> = {}
+
+    public async processInvocations(invocations: HogFunctionInvocation[]): Promise<HogFunctionInvocationResult[]> {
+        const results = await this.runManyWithHeartbeat(invocations, (item) => this.executePluginInvocation(item))
+
+        await this.processInvocationResults(results)
+        await this.updateJobs(results)
+        await this.produceQueuedMessages()
```

Comment:
question: Aren't those three calls already in `CdpCyclotronWorker.processBatch`?

Reply:
Ah you're right!